data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/PeerForwarder.java

```diff
@@ -24,20 +27,36 @@
 
 @DataPrepperPlugin(name = "peer_forwarder", type = PluginType.PROCESSOR)
 public class PeerForwarder extends AbstractPrepper<Record<ExportTraceServiceRequest>, Record<ExportTraceServiceRequest>> {
+    public static final String FORWARD_REQUEST_LATENCY_PREFIX = "forwardRequestLatency";
+    public static final String FORWARD_REQUEST_SUCCESS_PREFIX = "forwardRequestSuccess";
+    public static final String FORWARD_REQUEST_ERRORS_PREFIX = "forwardRequestErrors";
+    public static final String PEER_ENDPOINTS = "peerEndpoints";
+
     private static final Logger LOG = LoggerFactory.getLogger(PeerForwarder.class);
 
     private final HashRing hashRing;
     private final PeerClientPool peerClientPool;
     private final int maxNumSpansPerRequest;
 
+    private final Map<String, Timer> forwardRequestTimers;
+    private final Map<String, Counter> forwardedRequestCounters;
+    private final Map<String, Counter> forwardRequestErrorCounters;
+    private final PeerListProvider peerListProvider;
+
     public PeerForwarder(final PluginSetting pluginSetting,
+                         final PeerListProvider peerListProvider,
                          final PeerClientPool peerClientPool,
                          final HashRing hashRing,
                          final int maxNumSpansPerRequest) {
         super(pluginSetting);
         this.peerClientPool = peerClientPool;
         this.hashRing = hashRing;
         this.maxNumSpansPerRequest = maxNumSpansPerRequest;
+        this.peerListProvider = pluginMetrics.gauge(
+                PEER_ENDPOINTS, peerListProvider, provider -> (double) provider.getPeerList().size());
```

Comment:
I think we spoke about this in the slack channel, is it hard to provide the `pluginMetrics` as a reference to PeerListProviderFactory? So that both static and dynamic can emit metrics on their own like errors, refresh etc.

Reply:
ref #257